GliaWidgets/Sources/ViewController/Common/MediaUpgrade/MediaUpgradePresenter.swift

```diff
@@ -16,12 +16,25 @@ extension AlertPresenter {
         accepted: @escaping () -> Void,
         declined: @escaping () -> Void
     ) {
-        let alert = AlertViewController(
-            kind: .singleMediaUpgrade(conf,
-                                      accepted: accepted,
-                                      declined: declined),
-            viewFactory: viewFactory
-        )
-        present(alert, animated: true, completion: nil)
+        let completion = { [unowned self] in
+            let alert = AlertViewController(
+                kind: .singleMediaUpgrade(
+                    conf,
+                    accepted: accepted,
+                    declined: declined
+                ),
+                viewFactory: self.viewFactory
+            )
+            self.present(alert, animated: true)
+        }
+        guard let presented = self.presentedViewController else {
```

Comment:
Just a thought: is it possible to automate the logic of previous view dismissal by moving it into alert? Because currently such logic will be scattered all over the codebase.